CHANGELOG.md

```diff
@@ -6,6 +6,9 @@ Mapbox welcomes participation and contributions from everyone.
 #### Features
 #### Bug fixes and improvements
 - Fixed an issue where portions of alternative routes that overlap with the primary route weren't hidden correctly, suffering from precision problems. [#6228](https://github.com/mapbox/mapbox-navigation-android/pull/6228)
+- Fixed routes’ incorrect duration: now it relays on annotations if provided, otherwise on steps’s duration. [#6237](https://github.com/mapbox/mapbox-navigation-android/pull/6237)
```

Comment:
The fix doesn't change anything in `DirectionsRoute#duration`, it changes the `RouteProgress#distanceRemaining`. This entry still needs some work ahead of the release.

Reply:
@LukasPaczos what if
```suggestion
- Fixed routes’ incorrect `duration` entities in `AlternativeRouteMetadata` and `RouteProgress`: now it relies on annotations if provided, otherwise on steps’s duration. [#6237](https://github.com/mapbox/mapbox-navigation-android/pull/6237)
```

Reply:
```suggestion
- Fixed incorrect values in `AlternativeRouteInfo#duration` and `RouteProgress#durationRemaining`: now they rely on `route.leg.annotations.duration` if available, otherwise, `LegStep#duration` is used for calculations. [https://github.com/mapbox/mapbox-navigation-android/pull/6237](https://github.com/mapbox/mapbox-navigation-android/pull/6237)
```